packages/react-components/react-dialog/src/components/DialogSurface/useDialogSurface.ts

```diff
@@ -75,23 +75,24 @@ export const useDialogSurface_unstable = (
   }
 
   const { disableBodyScroll, enableBodyScroll } = useDisableBodyScroll();
-  const isBodyScrollLocked = Boolean(open && modalType !== 'non-modal');
+
   useIsomorphicLayoutEffect(() => {
-    if (isNestedDialog) {
+    if (isNestedDialog || modalType === 'non-modal') {
       return;
     }
-    if (open && isBodyScrollLocked && transitionStatus === 'entering') {
-      disableBodyScroll();
-    } else if (transitionStatus === 'exited') {
+
+    disableBodyScroll();
+
+    return () => {
       enableBodyScroll();
-    }
-  }, [disableBodyScroll, enableBodyScroll, isBodyScrollLocked, isNestedDialog, open, transitionStatus]);
+    };
+  }, [enableBodyScroll, isNestedDialog, disableBodyScroll, modalType]);
```

Comment:
Would it be possible to move this whole logic back into the `useDialog` hook? it was originally there, it was only moved to `useDialogSurface` to get access to the `transitionStatus`.

Reply:
As we discussed in offline, it makes sense to keep it there as we disable/enable scroll when `DialogSurface` gets mounted/unmounted.